File: Programmers/pro_42862_체육복.py
def solution(n, lost, reserve):
    clothes = [1]*n
    for i in lost:
        clothes[i-1]=0
        
    for j in range(len(reserve)):
        cur = reserve[j]
        if(clothes[cur-1]==0):#self
            clothes[cur-1]=1
            reserve[j]=1e9
            
    reserve.sort()
    for j in reserve:
        if(j==1e9):
            break
        if(j>=2 and clothes[j-2]==0):#left
            clothes[j-2]=1
        elif(j<=n-1 and clothes[j]==0):#right
            clothes[j]=1
    return sum(clothes)

File: Programmers/pro_42862_체육복.js
function solution(n, lost, reserve) {
    let clothes = Array.from({ length: n }, () => 1)
    for (let i in lost) {
        let lostIdx = lost[i] - 1
        clothes[lostIdx] -= 1
    }

    for (let i in reserve) {
        let reserveIdx = reserve[i] - 1
        if (clothes[reserveIdx] == 0) {
            clothes[reserveIdx] = 1
            reserve[i] = -1 // used
        }
    }

    reserve.sort().reverse()
    for (let i in reserve) {
        if (reserve[i] == -1)
            break
        let reserveIdx = reserve[i] - 1
        if (reserveIdx > 0 && clothes[reserveIdx - 1] == 0) {
            clothes[reserveIdx - 1] += 1
        } else if (reserveIdx < n - 1 && clothes[reserveIdx + 1] == 0) {
            clothes[reserveIdx + 1] += 1
        }
    }

    return clothes.filter(v => v == 1).length
}